src/Microsoft.DotNet.Build.Tasks/PackageFiles/publishtest.targets

```diff
@@ -65,16 +67,24 @@
       <Output TaskParameter="ResolvedCopyLocalItems" ItemName="TestCopyLocal" />
     </PrereleaseResolveNuGetPackageAssets>
 
-     <!-- We may have an indirect package reference that we want to replace with a project reference.
+    <!-- We may have an indirect package reference that we want to replace with a project reference.
           Those are part of RunTestsForProjectInputs. The order that we append to TestCopyLocal is
           significant, later entries override earlier entries. -->
     <ItemGroup>
       <TestCopyLocal Include="@(RunTestsForProjectInputs)" Exclude="@(PackagesConfigs)" />
     </ItemGroup>
 
+    <GetTargetArchitecture Condition="'$(TestWithLocalLibraries)'=='true'">
+      <Output TaskParameter="TargetArch" PropertyName="TargetArch" />
+    </GetTargetArchitecture>
+    <GetTargetOS Condition="'$(TestWithLocalLibraries)'=='true' AND '$(OSGroup)'=='AnyOS'">
+      <Output TaskParameter="TargetOS" PropertyName="TargetOS" />
+    </GetTargetOS>
+
     <!-- Test using locally built shim libraries. -->
     <ItemGroup Condition="'$(TestWithLocalLibraries)'=='true'">
-      <TestCopyLocal Include="$(BaseOutputPath)$(OSGroup).$(Platform).$(ConfigurationGroup)/Native/*.*" />
+      <TestCopyLocal Condition="'$(OSGroup)'!='AnyOS'" Include="$(BaseOutputPath)$(OSGroup).$(TargetArch).$(ConfigurationGroup)/Native/*.*" />
```

Comment:
We could be cross-building, and may not want to override the native libs.. Although for now, it's not a big deal, as the file extensions are different across Windows, Linux and OSX. This target just creates the test directory, and doesn't actually run tests like in test.targets, hence kept it so.